src/views/contact/ContactForm.js

```diff
@@ -0,0 +1,182 @@
+
+// @flow
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { hashHistory as history } from 'react-router';
+import FormInput from '../../components/contact/FormInput';
+import { labels, placeHolders, validateInput } from '../../utils/formUtils';
+import * as contactActions from '../../actions/contactActions';
+import utils from '../../utils/util';
+
+class ContactForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = ({
+      edit: !this.props.contact || !this.props.contact.id,
+      contact: this.props.contact,
+      originalContact: { ...this.props.contact },
+      errors: {},
+    });
+    // stupid create react app limitations
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleCancelEdit = this.handleCancelEdit.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleInput = this.handleInput.bind(this);
+    this.validate = this.validate.bind(this);
```

Comment:
Existuje ještě lepší (hezčí) způsob bindování metod objektu, dokázal bys přijít na to který?

Reply:
samozrejme, ale to by nemsmel byt pouzity create react app, ten omezuje pokrocile BABEL funkce jako transform-class-properties/
(https://github.com/facebookincubator/create-react-app/issues/167)
idealni je arrow zapis primo class/instance methody

Reply:
Imho by arrow fce v `create-react-app` měly fungovat, ale takhle to stačí.

Reply:
nefunguji, a je to zamer autoru create-react-app, kuli tomu ze tyto BABEL funkce jsou uz mene zazite, mene otestovane atd:
"The point of this project is to provide good defaults (i.e. sensible, tested and solid) with a good DX. We've had a discussion about including decorators by default (#107), but the current consensus is that with the spec changing we're not ready to include them just yet.

For now, you'll have to eject to add support for them!"

Reply:
V linkovaném issue je ale diskuze o dekorátorech, což je zase něco jiného. Pokud u svého PR zrušíš babel konfiguraci do původního stavu, budou ti arrow funkce fungovat :)

Reply:
urcite vim ze ejectem muzu delat co chci, mohl jsem zvolit i vlastni setup kompletne, ale drzel jsem se instrukci...zadani neni dokonael (viz regexp na isPhone ..) kazdopadne zminena diskuze se tyka vse babel preset stage 2, tady i transform class properties :-)